docs/data/stashcache/run-stashcache-container.md

```diff
@@ -0,0 +1,184 @@
+
+Which should output:
+
+```
+>Derek's first query!
+MPVSDSGFDNSSKTMKDDTIPTEDYEEITKESEMGDATKITSKIDANVIEKKDTDSENNITIAQDDEKVSWLQRVVEFFE
+```
+
+Readying for Production
+------------------------
+
+Additional configuration is needed to make StashCache production-ready.
+
+1. Add environment variables to set the name of the cache and the directory of the persistant cache.  It is important to remember that the directory set in the environment variable points to the directory **inside** the container.  It is later mapped to a host directory with the `--volume` option.
+
+An example final `docker run` command:
+```console
+user@host $ docker run --rm --publish <HOST PORT>:8000 \
+             --volume <HOST PARTITION>:/cache \
+             --env-file=/opt/xcache/.env \
+             opensciencegrid/stash-cache:fresh
+```
+
+Where the environment file on the docker host, `/opt/xcache/.env`, has the following contents:
+```file
+XC_RESOURCENAME=ProductionCache
+XC_ROOTDIR=/cache
+```
+
+It is recommended to use a container orchestration service such as [docker-compose](https://docs.docker.com/compose/) or [kubernetes](https://kubernetes.io/), or start the XCache container with systemd.
+
+An example systemd service file for xcache.  This will require creating the environment file in the directory `/opt/xcache/.env`. 
+
+!!! note
+    This example systemd file assumes <HOST PORT> is `8000` and  <HOST PARTITION> is `/srv/cache`.
+
+```file
+[Unit]
+Description=XCache Container
+After=docker.service
+Requires=docker.service
+
+[Service]
+TimeoutStartSec=0
+Restart=always
+ExecStartPre=-/usr/bin/docker stop %n
+ExecStartPre=-/usr/bin/docker rm %n
+ExecStartPre=/usr/bin/docker pull opensciencegrid/stash-cache:fresh
+ExecStart=/usr/bin/docker run --rm --name %n --publish 8000:8000 --volume /srv/cache:/cache --env-file /opt/xcache/.env opensciencegrid/stash-cache:fresh
+
+[Install]
+WantedBy=multi-user.target
+```
+
+This systemd file can be saved to `/etc/systemd/system/docker.stash-cache.service` and started with:
+
+```console
+root@host $ systemctl enable docker.stash-cache
+root@host $ systemctl start docker.stash-cache
+```
+
+!!! warning
+    [Register](../stashcache/install-cache/#registering-the-cache) the cache before considering it a production service.
+
+Additional Configuration (Optional)
+-----------------------------------
+
+In addition to the configuration avove before starting the container, write the following configuration on your docker host:
+
+1. Write in same file as above(`/opt/xcache/.env`) containing the following required environment  variables and values for your XCache:
+
+    - `XC_SPACE_HIGH_WM`: High watermark for disk usage;
+      when usage goes above the high watermark, the cache deletes until it hits the low watermark
+    - `XC_SPACE_LOW_WM`: Low watermark for disk usage;
+      when usage goes above the high watermark, the cache deletes until it hits the low watermark
+    - `XC_PORT`: TCP port that XCache listens on
```

Comment:
```suggestion
    - `XC_PORT`: TCP port that XCache listens on
```

Is this actually something useful to change in the container world? Who cares what port the service listens to in the container, we're going to munge it through Docker/Kubernetes/etc networking anyway. Just a thought.

Reply:
Ahh this reminds me  I need to add a line about this in optimization. IT will make more sense